File: .github/src/utils/wallet_utils.py
from model import Card, Money, Wallet
from model.common import CardType

def console_utils():
    menu = (
        "\n"
        "                                __           __                 \n"
        "  ____           ______   ____ |  | __ _____/  |_____________   \n"
        "_/ __ \\   ______ \\____ \\ /  _ \\|  |/ // __ \\   __\\_  __ \\__  \\  \n"
        "\\  ___/  /_____/ |  |_> >  <_> )    <\\  ___/|  |  |  | \\// __ \\_\n"
        " \\___  >         |   __/ \\____/|__|_ \\\\___  >__|  |__|  (____  /\n"
        "     \\/          |__|               \\/    \\/                 \\/ \n"
    )

    print(menu)

def menu():
    print("Welcome to e-poketra!")
    console_utils()
    print("1 - Put object in")
    print("2 - Put object out")
    print("3 - Get my balance")
    print("4 - Give my identifier")
    print("5 - Count object [Credit card, visit card, driving card, National Identity Card]")

def object_type_menu(wallet, object_id, pocket_number, obj_type):
    if obj_type == 1:
        amount = float(input("Money amount: "))
        wallet.put_object_in(pocket_number, Money(object_id, amount))
    elif obj_type == 3:
        wallet.put_object_in(pocket_number, Card(object_id, CardType.CREDIT_CARD))
    elif obj_type == 4:
        wallet.put_object_in(pocket_number, Card(object_id, CardType.VISIT_CARD))
    elif obj_type == 5:
        wallet.put_object_in(pocket_number, Card(object_id, CardType.DRIVING_CARD))
    elif obj_type == 6:
        wallet.put_object_in(pocket_number, Card(object_id, CardType.NI_CARD))
    else:
        print("Unrecognized object type")

def object_count_menu(wallet, obj_type):
    if obj_type == 1:
        print(f"Credit card count: {wallet.count_object(CardType.CREDIT_CARD)}")
    elif obj_type == 2:
        print(f"Visit card count: {wallet.count_object(CardType.VISIT_CARD)}")
    elif obj_type == 3:
        print(f"Driving card count: {wallet.count_object(CardType.DRIVING_CARD)}")
    elif obj_type == 4:
        print(f"NI card count: {wallet.count_object(CardType.NI_CARD)}")
    else:
        print("Unrecognized object type")

File: src/utils/wallet_utils.dart
import 'dart:io';

import '../model/Card.dart';
import '../model/Money.dart';
import '../model/Wallet.dart';

import '../model/common/card_type.dart';

class WalletUtils {
  static void consoleUtils() {
    String sigle = '''
                                __           __                 
  ____           ______   ____ |  | __ _____/  |_____________   
_/ __ \\   ______ \\____ \\ /  _ \\|  |/ // __ \\   __\\_  __ \\__  \\  
\\  ___/  /_____/ |  |_> >  <_> )    <\\  ___/|  |  |  | \\// __ \\_
 \\___  >         |   __/ \\____/|__|_ \\\\___  >__|  |__|  (____  /
     \\/          |__|               \\/    \\/                 \\/ 
''';

    print(sigle);
  }

  static void menu() {
    print('Welcome to e-poketra!');
    consoleUtils();
    print('1 - Put object in');
    print('2 - Put object out');
    print('3 - Get my balance');
    print('4 - Give my identifier');
    print('5 - Count object [Credit card, visit card, driving card, National Identity Card]');
  }

  static void objectTypeMenu(Wallet wallet, int objectId, int pocketNumber, int type) {
    switch (type) {
      case 1:
        stdout.write('Money amount : ');
        double amount = double.parse(stdin.readLineSync()!);
        wallet.putObjectIn(pocketNumber, Money(objectId, amount));
        break;
      case 2:
        break;
      case 3:
        wallet.putObjectIn(pocketNumber, Card(objectId, CardType.CREDIT_CARD));
        break;
      case 4:
        wallet.putObjectIn(pocketNumber, Card(objectId, CardType.VISIT_CARD));
        break;
      case 5:
        wallet.putObjectIn(pocketNumber, Card(objectId, CardType.DRIVING_CARD));
        break;
      case 6:
        wallet.putObjectIn(pocketNumber, Card(objectId, CardType.NI_CARD));
        break;
      default:
        print('Unrecognized object type');
    }
  }

  static void objectCountMenu(Wallet wallet, int type) {
    switch (type) {
      case 1:
        print('Credit card count : ${wallet.countObject(CardType.CREDIT_CARD)}');
        break;
      case 2:
        print('Credit card count : ${wallet.countObject(CardType.VISIT_CARD)}');
        break;
      case 3:
        print('Credit card count : ${wallet.countObject(CardType.DRIVING_CARD)}');
        break;
      case 4:
        print('Credit card count : ${wallet.countObject(CardType.NI_CARD)}');
        break;
      default:
        print('Unrecognized object type');
    }
  }
}
